### DISCUSSION TITLE
Await JS

### DISCUSSION THREAD

**Discussion by emargareten** [livewire/livewire#5985] [3. Feature ideas]
I wonder if it would be possible to get the javascript result in the component (or the promise result in case of a promise)? for example:
```php
$confirmed = $this->awaitJs("confirm('Are you sure?')")
```

**Comment by vlados** [comment]
I think it's only possible via events. This approach for me is impossible due to the fact that php is runned only once

**Reply by emargareten** [reply]
I was thinking of having it possible through some kind of hack (like all of livewire...)

**Comment by laserhybiz** [comment]
Something like this should work

- Cache some generated random key with an empty value
- Return the JS to run with the generated key
- From the JS side submit the result to some route that will update the value for this key
- In PHP run a while loop until this key has a value (with a timeout option)

**Comment by flying-elephant-git** [comment]
Wouldn't it be better to use callbacks? I've solved something similar with a Trait "HandlesCallbacks":

`<?php

namespace App\Livewire\Traits;

use Illuminate\Support\Arr;

trait HandlesCallbacks
{
    protected function flushCallbacks()
    {
        $now = time();
        session()->put('callbacks', Arr::where(session()->get('callbacks', []), fn ($callback) => $callback['expires'] > $now));
    }

    protected function callback(?callable $callback, int $minutes = 2): string
    {
        $this->flushCallbacks();

        $callbackId = uniqid('callback_');

        if (is_callable($callback)) {
            session()->put('callbacks.'.$callbackId, ['callback' => $callback, 'expires' => time() + ($minutes * 60)]);
        }

        return $callbackId;
    }

    public function runCallback(string $callbackId)
    {
        $this->flushCallbacks();

        $callback = session()->remove('callbacks.'.$callbackId);

        if ($callback && is_callable($callback['callback'])) {
            $callback['callback']();
        }
    }
}

`

And then in your component:

`
$callbackId = $this->callback([$this, 'someFunction']);
$this->js('confirm("...")&& $wire.' . $callbackId . '()');
`
